File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/HomeContract.java
package com.android.learn.mcontract;

import com.android.learn.base.mmodel.BannerListData;
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData;
import com.android.learn.base.mmodel.FeedArticleListData;
import com.android.learn.base.mview.BaseView;

/**
 * Created by gaolei on 2018/6/18.
 */

public class HomeContract {

    public interface Presenter {

        void getBannerInfo();

        void getFeedArticleList(int num);

        void onLoadMore();

        void onRefreshMore();

        void addCollectArticle(int position, FeedArticleData feedArticleData);

        void cancelCollectArticle(int position, FeedArticleData feedArticleData);
    }

    public interface View extends BaseView {

        void showArticleList(FeedArticleListData itemBeans, boolean isRefresh);

        void showBannerList(BannerListData itemBeans);

        void showCollectArticleData(int position, FeedArticleData feedArticleData);

        void showCancelCollectArticleData(int position, FeedArticleData feedArticleData);
        void showCancelCollectArticleData(int id);
    }
}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/HomeContract.kt
package com.android.learn.mcontract

import com.android.learn.base.mmodel.BannerListData
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData
import com.android.learn.base.mmodel.FeedArticleListData

/**
 * Created by gaolei on 2018/6/18.
 */

class HomeContract {

    interface Presenter {

        fun getBannerInfo()

        fun getFeedArticleList(num: Int)

        fun onLoadMore()

        fun onRefreshMore()

        fun addCollectArticle(position: Int, feedArticleData: FeedArticleData)

        fun cancelCollectArticle(position: Int, feedArticleData: FeedArticleData)
    }

    interface View {

        fun showArticleList(itemBeans: FeedArticleListData, isRefresh: Boolean)

        fun showBannerList(itemBeans: BannerListData)

        fun showCollectArticleData(position: Int, feedArticleData: FeedArticleData)

        fun showCancelCollectArticleData(position: Int, feedArticleData: FeedArticleData)
        fun showCancelCollectArticleData(id: Int)
    }
}
